src/rawArray/ops.ts

```diff
@@ -0,0 +1,114 @@
+    // This translates "...", ":", null, etc into a list of slices.
+    const normalizedDstSelection = normalizeArraySelection(dstSelection, dstShape, true);
+    // Above we force the results to be dstSliceIndices only, without integer selections making this cast is safe.
+    const [dstSliceIndices] = selectionToSliceIndices(normalizedDstSelection, dstShape);
+
+    const normalizedSourceSelection = normalizeArraySelection(sourceSelection, sourceShape, false);
+    const [sourceSliceIndicies] = selectionToSliceIndices(normalizedSourceSelection, sourceShape);
+
+    _setRawArrayDirect(dstArr, dstStrides, dstSliceIndices as SliceIndices[], sourceArr, sourceStrides, sourceSliceIndicies);
+}
+
+function _setRawArrayDirect(dstArr: TypedArray, dstStrides: number[], dstSliceIndices: SliceIndices[], sourceArr: TypedArray, sourceStrides: number[], sourceSliceIndicies: (SliceIndices | number)[]) {
+    if (sourceSliceIndicies.length === 0) {
+        // Case when last source dimension is squeezed
+        dstArr.set(sourceArr);
+        return;
+    }
+
+    // Get current indicies and strides for both destination and source arrays
+    const [currentDstSlice, ...nextDstSliceIndicies] = dstSliceIndices;
+    const [currentSourceSlice, ...nextSourceSliceIndicies] = sourceSliceIndicies;
+
+    const [currentDstStride, ...nextDstStrides] = dstStrides;
+    const [currentSourceStride, ...nextSourceStrides] = sourceStrides;
+
+    // This source dimension is squeezed
+    if (typeof currentSourceSlice === "number") {
+        /*
+        Sets dimension offset for squeezed dimension.
+
+        Ex. if 0th dimension is squeezed to 2nd index (numpy : arr[2,i])
+
+            sourceArr[stride[0]* 2 + i] --> sourceArr.subarray(stride[0] * 2)[i] (sourceArr[i] in next call)
+
+        Thus, subsequent squeezed dims are appended to the source offset.
+        */
+        _setRawArrayDirect(
+            // Don't update destination offset/slices, just source
+            dstArr, dstStrides, dstSliceIndices,
+            sourceArr.subarray(currentSourceStride * currentSourceSlice),
+            nextSourceStrides,
+            nextSourceSliceIndicies,
+        );
+        return;
+    }
+
+    const [from, _to, step, outputSize] = currentDstSlice; // just need start and size
+    const [sfrom, _sto, sstep, _soutputSize] = currentSourceSlice; // Will always be subset of dst, so don't need output size just start
+
+    if (dstStrides.length === 1 && sourceStrides.length === 1) {
+        if (step === 1 && currentDstStride === 1 && sstep === 1 && currentSourceStride === 1) {
+            dstArr.set(sourceArr.subarray(sfrom, sfrom + outputSize), from);
+            return;
+        }
+
+        for (let i = 0; i < outputSize; i++) {
+            dstArr[currentDstStride * (from + (step * i))] = sourceArr[currentSourceStride * (sfrom + (sstep * i))];
```

Comment:
Hats off for getting this correct, it took me a while for NestedArrays and I think those were actually easier

Reply:
There was a bit of mental overhead with figuring out what you implemented. But thanks for that! No way I would have been able to otherwise probably, and I'm quite happy with the final result :) Learned a lot about `TypedArray`s and views of the underlying data buffers.